src/main/java/roomescape/reservation/service/ReservationService.java

```diff
@@ -69,13 +73,29 @@ public List<ReservationResponse> getReservations(
     }
 
     public List<MyReservationResponse> getMyReservations(final LoginMember loginMember) {
-        final List<Reservation> reservations = reservationRepository.findAllByMemberIdOrderByDateDesc(loginMember.id());
-        return reservations.stream()
+        final List<MyReservationResponse> reservations = reservationRepository.findAllByMemberIdOrderByDateDesc(
+                        loginMember.id()).stream()
                 .map(MyReservationResponse::new)
                 .toList();
+        final List<MyReservationResponse> waitings = waitingRepository.findWaitingWithRankByMemberId(
+                        loginMember.id()).stream()
+                .map(MyReservationResponse::new)
+                .toList();
+        final ArrayList<MyReservationResponse> responses = new ArrayList<>(reservations);
+        responses.addAll(waitings);
+        return responses;
     }
 
     public void cancelReservationById(final long id) {
+        final Optional<Waiting> waiting = waitingRepository.findFirstByReservationIdOrderByCreatedAtAsc(id);
+        if (waiting.isPresent()) {
+            Waiting waiting2 = waiting.get();
```

Comment:
존재하면 가져와서 처리하고 없으면 아무 예외없이 종료하는 방식은 정상적으로 처리가 된 것인지 아닌지 오해할 여지가 있다고 생각해요. 🤔

예약 대기를 조회했을 경우 존재하지 않으면 바로 예외를 던져주는 것은 어떨까요? 
```suggestion
        Waiting waiting = waitingRepository.findFirstByReservationIdOrderByCreatedAtAsc(id)
            .orElseThrow(() -> new BadRequestException("예약을 찾을 수 없습니다."));
```

Reply:
이 로직은 아래와 같은 기능을 수행합니다.
- 삭제하고자 하는 예약에 **대기가 없는 경우** -> 예약을 삭제한다
- 삭제하고자 하는 예약게 **대기가 있는 경우** -> 기존 예약 정보를 대기 정보로 교체하고, 예약은 삭제하지 않는다
즉 대기가 없는 경우에도 예외가 발생하지 않아야 합니다!

대기가 예약을 참고하는 구조 때문에 메서드가 예상과 다르게 행동하여 혼란을 가져다 주었네요ㅎㅎ
리팩터링을 하면서 `ifPresentOrElse()` 메서드를 사용하여 혼동을 최소한으로 하였습니다

0995fe2b